aircmd/core/artifact.py

```diff
@@ -13,41 +15,42 @@
 
 class BuildCommand(ClickCommandMetadata):
     command_name: str = "build"
-    command_help: str = "Build aircmd"
+    command_help: str = "Builds aircmd"
 
-@core_group.command(BuildCommand())
-@pass_pipeline_context
-async def build(ctx: PipelineContext, settings: GlobalSettings) -> None:
-    """Build aircmd"""
-    async with ctx.dagger_connection as client:
-        runner = (
-            client.container(platform=dagger.Platform("linux/amd64"))
-            .from_("python:3.11-buster")
-            .with_directory("/app", client.host().directory(".", include=["./pyproject.toml", "./poetry.lock"]))
-            .with_workdir("/app")
-            .with_exec(["poetry", "install"]) # cache the installation of dependencies so we don't rerun every time we change src
-            .with_directory("/app", client.host().directory(".", include=["./actions", "./aircmd", "./pyproject.toml", "./poetry.lock"]))
+async def build_task(parent_pipeline: Optional[Pipeline] = None) -> None:
+    pipeline = await Pipeline.create("build", parent_pipeline=parent_pipeline)
+    mypy_cache: CacheVolume = pipeline.dagger_client.cache_volume("pip_cache")
+
+    ctr = (with_poetry(pipeline)
+            .with_directory("/src", pipeline.dagger_client.host().directory(".", include=["./aircmd", "./pyproject.toml", "./poetry.lock"]))
+            .with_workdir("/src")
+            .with_mounted_cache("/src/.mypy_cache", mypy_cache)
+            .with_exec(["poetry", "install"])
             .with_exec(["poetry", "run", "mypy", "."])
             .with_exec(["poetry", "run", "ruff", "."])
             .with_exec(["poetry", "build"])
-        )
-        await runner.stdout()
-
-
+    )
 
-class TestCommand(ClickCommandMetadata):
-    command_name: str = "test"
-    command_help: str = "Run tests for aircmd"
-    arguments: List[ClickArgument] = [ClickArgument(name="query", required=False)]
+    await ctr.stdout()
 
-
-class PublishCommand(ClickCommandMetadata):
-    command_name: str = "publish"
-    command_help: str = "Publish aircmd to PyPI"
-    arguments: List[ClickArgument] = [ClickArgument(name="query", required=False)]
+    #output = await ctr.stdout()
+    #print(output[:300])
+    return None
 
 
 class CICommand(ClickCommandMetadata):
     command_name: str = "ci"
     command_help: str = "Run CI for aircmd"
-    arguments: List[ClickArgument] = [ClickArgument(name="query", required=False)]
+
+@core_group.command(BuildCommand())
+@pass_pipeline_context
+async def build(ctx: PipelineContext) -> None:
+    await ctx.run_pipelines([build_task()], concurrency=1)
```

Comment:
I see. I agree that it does add some complexity. The reason I set it up like this is so that the end user can have full control over any validation of inputs while still ensuring that we can enforce typing in the centralized location of `aircmd`. In other words, they can do whatever they want to validate arguments with pydantic at runtime with this approach, and the entire interface of what pydantic can give them is exposed, vs using kwargs where one doesn't necessarily get that easy access to the pydantic model.

These are perhaps simplistic examples in this PR, but I could imagine the end user wanting to validate arguments, options etc beyond what, say, typer might offer them and it felt like giving them the pydantic model itself was the most powerful way to do that. Then they simply implement whatever their own validations are necessary on top of what `aircmd` requires

It feels like it would be more ergonomic to either make accessing/using the model opt-in (perhaps a get_model() function for the command) or alternatively giving people the tools to perform their own marshalling to a pydantic model. It's something I want to revisit, especially looking at how dagger do project entrypoints works. But will leave out of the scope of this PR